### PR TITLE
[Tizen] Add libwebsockets and support C++11

### PR DESCRIPTION
**Pull request by pyrasis**
- Add libwebsockets header and x86 library.
- Support C++11 (Tizen SDK contains gcc 4.5. it doesn't support nullptr)

### CODE REVIEW

File: CocosDenshion/proj.tizen/.cproject [hunk 1]
```diff
@@ -24,12 +24,12 @@
 							<targetPlatform binaryParser="org.eclipse.cdt.core.ELF" id="org.tizen.nativeide.target.sbi.gnu.platform.base.1067387441" osList="linux,win32" superClass="org.tizen.nativeide.target.sbi.gnu.platform.base"/>
 							<builder autoBuildTarget="all" buildPath="${workspace_loc:/cocosdenshion/Debug-Tizen-Emulator}" enableAutoBuild="true" id="org.tizen.nativecpp.target.sbi.gnu.builder.1261707379" keepEnvironmentInBuildfile="false" managedBuildOn="true" name="Tizen Application Builder" superClass="org.tizen.nativecpp.target.sbi.gnu.builder"/>
 							<tool command="i386-linux-gnueabi-ar.exe" commandLinePattern="${COMMAND} ${FLAGS} ${OUTPUT_FLAG} ${OUTPUT_PREFIX}${OUTPUT} ${INPUTS}" id="org.tizen.nativecpp.tool.sbi.gnu.archiver.689316005" name="Archiver" superClass="org.tizen.nativecpp.tool.sbi.gnu.archiver"/>
-							<tool command="clang++.exe" id="org.tizen.nativecpp.tool.sbi.gnu.cpp.compiler.1533358351" name="C++ Compiler" superClass="org.tizen.nativecpp.tool.sbi.gnu.cpp.compiler">
+							<tool command="i386-linux-gnueabi-g++.exe" id="org.tizen.nativecpp.tool.sbi.gnu.cpp.compiler.1533358351" name="C++ Compiler" superClass="org.tizen.nativecpp.tool.sbi.gnu.cpp.compiler">
```
Comment (dumganhar) [inline CocosDenshion/proj.tizen/.cproject:27]: Why not use clang++? clang++ seems to support c++11 better.

Comment (pyrasis) [inline CocosDenshion/proj.tizen/.cproject:27]: this is clang++ compile error  message.

```
clang++.exe -I"pch" -D_DEBUG -DTIZEN -I"../../../cocos2dx" -I"../../../cocos2dx/include" -I"../../../cocos2dx/kazmath/include" -I"../../../cocos2dx/platform/tizen" -I"../../../cocos2dx/platform/third_party/tizen/rootstraps/tizen-emulator-2.1.native/usr/include" -I"../../../cocos2dx/platform/third_party/tizen/rootstraps/tizen-emulator-2.1.native/usr/include/freetype2" -I"../../../cocos2dx/platform/third_party/tizen/prebuilt/libwebp/include" -O0 -g3 -Wall -c -fmessage-length=0 -std=c++0x -target i386-tizen-linux-gnueabi -gcc-toolchain C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/ -ccc-gcc-name i386-linux-gnueabi-g++ -march=i386 -Wno-gnu -fPIC --sysroot="C:/tizen-sdk/platforms/tizen2.1/rootstraps/tizen-emulator-2.1.native" -I"C:/tizen-sdk/platforms/tizen2.1/rootstraps/tizen-emulator-2.1.native/usr/include/libxml2" -I"C:\tizen-sdk\library" -I"C:/tizen-sdk/platforms/tizen2.1/rootstraps/tizen-emulator-2.1.native/usr/include" -I"C:/tizen-sdk/platforms/tizen2.1/rootstraps/tizen-emulator-2.1.native/usr/include/osp" -D_APP_LOG -MMD -MP -MF"src/tilemap_parallax_nodes/CCTMXXMLParser.d" -MT"src/tilemap_parallax_nodes/CCTMXXMLParser.d" -o "src/tilemap_parallax_nodes/CCTMXXMLParser.o" "C:/Users/pyrasis/Desktop/cocos2d-x_fork/cocos2dx/tilemap_parallax_nodes/CCTMXXMLParser.cpp"
In file included from C:/Users/pyrasis/Desktop/cocos2d-x_fork/cocos2dx/tilemap_parallax_nodes/CCTMXXMLParser.cpp:28:
In file included from C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/lib/gcc/i386-linux-gnueabi/4.5.4/../../../../i386-linux-gnueabi/include/c++/4.5.4\map:59:
C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/lib/gcc/i386-linux-gnueabi/4.5.4/../../../../i386-linux-gnueabi/include/c++/4.5.4\bits/stl_tree.h:136:4: error: call to implicitly-deleted copy constructor of 'std::pair<const std::basic_string<char>, std::basic_string<char> >'
          _M_value_field(std::forward<_Args>(__args)...) { }
          ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/lib/gcc/i386-linux-gnueabi/4.5.4/../../../../i386-linux-gnueabi/include/c++/4.5.4\ext/new_allocator.h:111:23: note: in instantiation of function template specialization 'std::_Rb_tree_node<std::pair<const std::basic_string<char>, std::basic_string<char> > >::_Rb_tree_node<const std::pair<const std::basic_string<char>, std::basic_string<char> > &>' requested here
        { ::new((void *)__p) _Tp(std::forward<_Args>(__args)...); }
                             ^
C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/lib/gcc/i386-linux-gnueabi/4.5.4/../../../../i386-linux-gnueabi/include/c++/4.5.4\bits/stl_tree.h:394:32: note: in instantiation of function template specialization '__gnu_cxx::new_allocator<std::_Rb_tree_node<std::pair<const std::basic_string<char>, std::basic_string<char> > > >::construct<const std::pair<const std::basic_string<char>, std::basic_string<char> > &>' requested here
              _M_get_Node_allocator().construct(__tmp,
                                      ^
C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/lib/gcc/i386-linux-gnueabi/4.5.4/../../../../i386-linux-gnueabi/include/c++/4.5.4\bits/stl_tree.h:899:24: note: in instantiation of function template specialization 'std::_Rb_tree<std::basic_string<char>, std::pair<const std::basic_string<char>, std::basic_string<char> >, std::_Select1st<std::pair<const std::basic_string<char>, std::basic_string<char> > >, std::less<std::basic_string<char> >, std::allocator<std::pair<const std::basic_string<char>, std::basic_string<char> > > >::_M_create_node<const std::pair<const std::basic_string<char>, std::basic_string<char> > &>' requested here
      _Link_type __z = _M_create_node(__v);
                       ^
C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/lib/gcc/i386-linux-gnueabi/4.5.4/../../../../i386-linux-gnueabi/include/c++/4.5.4\bits/stl_tree.h:1191:34: note: in instantiation of member function 'std::_Rb_tree<std::basic_string<char>, std::pair<const std::basic_string<char>, std::basic_string<char> >, std::_Select1st<std::pair<const std::basic_string<char>, std::basic_string<char> > >, std::less<std::basic_string<char> >, std::allocator<std::pair<const std::basic_string<char>, std::basic_string<char> > > >::_M_insert_' requested here
            return pair<iterator, bool>(_M_insert_(__x, __y, __v), true);
                                        ^
C:/tizen-sdk/tools/smart-build-interface/../i386-linux-gnueabi-gcc-4.5/lib/gcc/i386-linux-gnueabi/4.5.4/../../../../i386-linux-gnueabi/include/c++/4.5.4\bits/stl_map.h:501:21: note: in instantiation of member function 'std::_Rb_tree<std::basic_string<char>, std::pair<const std::basic_string<char>, std::basic_string<char> >, std::_Select1st<std::pair<const std::basic_string<char>, std::basic_string<char> > >, std::less<std::basic_string<char> >, std::allocator<std::pair<const std::basic_string<char>, std::basic_string<char> > > >::_M_insert_unique' requested here
      { return _M_t._M_insert_unique(__x); }
                    ^
C:/Users/pyrasis/Desktop/cocos2d-x_fork/cocos2dx/tilemap_parallax_nodes/CCTMXXMLParser.cpp:327:28: note: in instantiation of member function 'std::map<std::basic_string<char>, std::basic_string<char>, std::less<std::basic_string<char> >, std::allocator<std::pair<const std::basic_string<char>, std::basic_string<char> > > >::insert' requested here
            attributeDict->insert(pair<std::string, std::string>(key, value));
src/tilemap_parallax_nodes/subdir.mk:61: recipe for target 'src/tilemap_parallax_nodes/CCTMXXMLParser.o' failed
```

Comment (dumganhar) [inline CocosDenshion/proj.tizen/.cproject:27]: No idea about that. :(

**Comment by dumganhar** [comment]
Thanks. Merged.

**Pull request merged**